ocaml/idl/datamodel_repository.ml

```diff
@@ -223,8 +246,15 @@ let t =
           "The file name of the GPG public key of this repository"
       ; field ~qualifier:StaticRO ~lifecycle:[] ~ty:origin "origin"
           ~default_value:(Some (VEnum "remote"))
-          "The origin of the repository. 'remote' if the origin of the \
+          "The origin of this repository. 'remote' if the origin of the \
            repository is a remote one, 'bundle' if the origin of the \
-           repository is a local bundle file."
+           repository is a local bundle file, 'remote_pool' if the origin of \
+           the repository is a remote pool"
+      ; field ~qualifier:StaticRO ~lifecycle:[] ~ty:String
+          ~default_value:(Some (VString "")) "address"
```

Comment:
The `Repository.address` and `Repository.certificate` sound not great. How about reusing `Repository.binary_url` and `Repository.remote_certificate`?

Reply:
I think we will need to use `Stunnel.with_client_proxy` to reposync from the remote pool, as the certificate of the remote pool might be self-signed, and reposync directly from it will fail when checking the self-signed remote certificate. So the `binary_url` will be like: "http://127.0.0.1:yum_repo_port/repository/repository_name/"

Reply:
And I don't add the prefix `remote_` as I think although this attribute is only valid for `remote_pool` now, while it is an attribute of the repository, just like `name`, `binary_url`.

Reply:
It's not clear to me for what a `Repository.address` stands for an update repository. While the `Repository.binary_url` is a clear name of a YUM repository. The url like `http://127.0.0.1 ...` is only an internal implementation which should not be in the configuration. And actually a URL like "https://<remote-pool-coordinator-fqdn>/applied_updates" is a general interface of the remote pool repository. This URL would have to be configured in the stunnel client. Note that the stunnel client is an internal implementation also. So from the interface's point of view, the URL is the really the Repository.binary_url.

Reply:
If we need to add a new HTTP API "/applied_updates", then I think I need to update the design and update the groomed ticket first.